external-builds/pytorch/run_linux_pytorch_tests.py

```diff
@@ -0,0 +1,378 @@
+'All' is also possible. Then all skip tests for all pytorch versions are included.
+If no PyTorch version is given, it is auto-determined by the PyTorch used to run pytest.""",
+    )
+
+    env_root_dir = os.getenv("THEROCK_ROOT_DIR")
+    parser.add_argument(
+        "--the-rock-root-dir",
+        default=env_root_dir if env_root_dir is not None else "",
+        required=False,
+        help="""Overwrites the root directory of TheRock.
+By default TheRock root dir is determined based on this script's location.""",
+    )
+
+    parser.add_argument(
+        "--debug",
+        default=False,
+        required=False,
+        action=argparse.BooleanOptionalAction,
+        help="""Inverts the selection. Only runs skipped tests.""",
+    )
+
+    parser.add_argument(
+        "-k",
+        default="",
+        required=False,
+        help="""Overwrites the pytest -k option that decides which tests should be run or skipped""",
+    )
+
+    parser.add_argument(
+        "--no-cache",
+        default=False,
+        required=False,
+        action=argparse.BooleanOptionalAction,
+        help="""Disable pytest caching. Useful when only having read-only access to pytorch directory""",
+    )
+
+    args = parser.parse_args(argv)
+    return args
+
+
+def detect_amdgpu_family(amdgpu_family: str = "") -> list[str]:
+    """Detect and configure AMDGPU family using amdgpu-arch command.
+
+    This function always queries amdgpu-arch to get available GPUs and sets
+    HIP_VISIBLE_DEVICES to select the appropriate GPU(s) for testing.
+
+    Args:
+        amdgpu_family: AMDGPU family string. Can be:
+            - Empty string (default): Auto-detect and use first GPU (index 0)
+            - Specific arch (e.g., "gfx1151"): Find and use matching GPU
+            - Wildcard family (e.g., "gfx94X"): Find all matching GPUs
+
+    Returns:
+        List of detected AMDGPU family strings. Exits on failure.
+
+    Side effects:
+        Sets HIP_VISIBLE_DEVICES environment variable to comma-separated GPU indices.
+    """
+    try:
+        # Find amdgpu-arch executable in current python environment
+        print(
+            f"Searching for amdgpu-arch in subdirectories of {Path(sys.executable).parent.parent}"
+        )
+        proc = subprocess.run(
+            ["find", Path(sys.executable).parent.parent, "-name", "amdgpu-arch"],
+            capture_output=True,
+            text=True,
+            check=False,
+        )
+        # There might be 2 matches: rocm_sdk_core and rocm_sdk_devel, so just take the first one
+        amdgpu_arch_cmd = proc.stdout.split("\n")[0].strip()
+
+        if not amdgpu_arch_cmd:
+            print("[ERROR] amdgpu-arch command not found in Python environment")
+            sys.exit(1)
+
+        # Query available GPUs
+        proc = subprocess.run(
+            [amdgpu_arch_cmd], capture_output=True, text=True, check=False
+        )
```

Comment:
I _think_ the `amdgpu-arch` console script from the current venv should take precedence over any system installs of PATH, so simply running `subprocess.run("amdgpu-arch")` may work reliably enough.

You could also use `shutil.which` instead of `find`: https://docs.python.org/3/library/shutil.html#shutil.which. Note that on Windows, there is a `find.exe` in System32 that is different from the bash `find`.
* I actually alias `find` in my environment so scripts like this work for me, but most users will not have already done that, `find=D:\tools\cmder\vendor\git-for-windows\usr\bin\find.exe $*` (in the `config/user_aliases.cmd` for https://cmder.app/)

Reply:
Ah, as of https://github.com/ROCm/TheRock/commit/52810dabaf56096a8c0b750c1f84c43482ad6266 it should be called `offload-arch`

https://github.com/ROCm/TheRock/blob/5afb2d02a5cb18f39816fbc36ba3c619a9c5c7d7/build_tools/packaging/python/templates/rocm-sdk-core/src/rocm_sdk_core/_cli.py#L66-L67